lib/offline/SQLiteWrapper.hpp

```diff
@@ -273,6 +285,25 @@ namespace MAT_NS_BEGIN {
             return true;
         }
 
+        void shutdown_sqlite()
+        {
+            if (!m_skipInitAndShutdown)
+            {
+                if (m_initAndShutdownLock && m_instanceCount)
+                {
+                    LOCKGUARD(*m_initAndShutdownLock);
+                    *m_instanceCount -= 1;
+                    if (*m_instanceCount <= 0)
```

Comment:
It feels a bit odd to see this check for `less than zero`. What are the preconditions for that? `FlushAndTeardown` should be rather safe, I believe it's already ref-counting properly.

Reply:
Why exactly do you first decrement it unconditionally, and then check if it turned out to be less than zero?

Reply:
Good catch. I think I originally did this out of a defensive habit, just in case the value isn't what was expected, but I updated this logic to be a bit clearer, and I changed it so shutdown only gets called when the count is exactly 1 (since the safer thing is to not shutdown multiple times)